src/main.zig

```diff
@@ -80,18 +115,56 @@ fn handleBuiltin(argv: [][]const u8, ally: *std.mem.Allocator) !bool {
                 }
                 return true;
             };
-            try cd(home);
+            defer ally.free(home);
+            try cd(ally, home);
             return true;
         } else {
-            std.debug.assert(argv.len == 2);
-            try cd(argv[1]);
+            std.debug.assert(operands.items.len == 1);
+            try cd(ally, operands.items[0]);
             return true;
         }
     }
+
+    if (std.mem.eql(u8, argv[0], "export")) {
+        if (argv.len == 1) {
+            for (std.os.environ) |envvar| {
+                try stdout.print("{s}\n", .{envvar});
+            }
+        } else {
+            const eql_position = std.mem.indexOf(u8, argv[1], "=");
+            if (eql_position) |pos| {
+                const name = argv[1][0..pos];
+                const word = argv[1][pos + 1 ..];
+                try envExport(ally, name, word);
+            } else {
+                try shigError("cd: TODO export existing variables", .{});
```

Comment:
ah, std.process.getEnvMap seems to be exactly what I have been looking for all night but couldn't find. I really wasn't happy with the current solution so that should hopefully help. It does not seem to update os.environ though so I should probably create a global object somehow.

Will fix that typo :+1:

Reply:
the comment should be resolved